utility/firebase.js

```diff
@@ -310,3 +311,55 @@ export const subscribeToCMSStatus = (dateCallback) => {
       dateCallback(offUntilDate);
     });
 };
+
+export const getActiveHackathon = db
+  .collection(InternalWebsitesCollection)
+  .doc(LivesiteCollection)
+  .get()
+  .then((doc) => doc.data()?.activeHackathon);
+
+export const subscribeToLivesiteAnnouncements = (hackathon, callback) => {
+  return db
+    .collection(Hackathons)
+    .doc(hackathon)
+    .collection('Announcements')
+    .orderBy('timestamp', 'desc')
+    .onSnapshot((querySnapshot) => {
+      const announcements = {};
+      querySnapshot.docs.forEach((doc) => {
+        announcements[doc.id] = doc.data();
+      });
+      callback(announcements);
+    });
+};
+
+export const updateAnnouncement = async (hackathon, announcement) => {
+  if (announcement.id) {
+    const ref = db
+      .collection(Hackathons)
+      .doc(hackathon)
+      .collection('Announcements')
+      .doc(announcement.id);
+    delete announcement.id;
```

Comment:
why is the `id` being deleted here?

Reply:
We don't need to store the id in the object itself since that's the key in firebase